bionic/task_state.py

```diff
@@ -142,13 +145,33 @@ def attempt_to_complete_from_cache(self):
         if not all(axr.can_load() for axr in self._cache_accessors):
             return
 
-        # We only load the hashed result while completing task state
-        # and lazily load the entire result when needed later.
-        value_hashes_by_dnode = {}
-        for accessor in self._cache_accessors:
-            value_hash = accessor.load_result_value_hash()
-            value_hashes_by_dnode[accessor.query.dnode] = value_hash
-        self._result_value_hashes_by_dnode = value_hashes_by_dnode
+        self._load_value_hashes()
+
+    def refresh_all_cache_state(self, bootstrap):
+        """
+        Refreshes all state that depends on the persistent cache.
+
+        This is useful if the external cache state might have changed since we last
+        worked with this task.
+        """
+
+        # If this task state is not initialized or not persisted, there's nothing to
+        # refresh.
+        if not self._is_initialized or not self.should_persist:
+            return
+
+        self.refresh_cache_accessors(bootstrap)
+
+        # If we haven't loaded anything from the cache, we can stop here.
+        if self._result_value_hashes_by_dnode is None:
+            return
+
+        # Otherwise, let's update our value hashes from the cache.
+        if all(axr.can_load() for axr in self._cache_accessors):
+            self._load_value_hashes()
+        else:
+            self._result_value_hashes_by_dnode = None
+            self.is_complete = False
```

Comment:
You don't need to do this, we don't have `TaskState.is_complete` anymore.

Reply:
Nice, thanks!